examples/lessons/09/app.js

```diff
@@ -1,82 +1,100 @@
+    addControls();
+    addKeyboardHandler(canvas);
 
-  onInitialize: ({gl, canvas}) => {
+    gl.clearColor(0, 0, 0, 1);
+    gl.clearDepth(1);
     gl.blendFunc(gl.SRC_ALPHA, gl.ONE);
     gl.enable(gl.BLEND);
+//    gl.disable(gl.DEPTH_TEST);
 
-    var scene = new Group(gl);
-
-    addEvents(canvas, {
-      onKeyDown(e) {
-        switch (e.key) {
-        case 'up':
-          tilt -= 1.5;
-          break;
-        case 'down':
-          tilt += 1.5;
-          break;
-        // handle page up/down
-        default:
-          if (e.code === 33) {
-            zoom -= 0.1;
-          } else if (e.code === 34) {
-            zoom += 0.1;
-          }
-        }
-      }
-    });
-
-    loadTextures(gl, {
-      urls: ['star.gif'],
-      mipmap: true,
-      parameters: [{
-        [gl.TEXTURE_MAG_FILTER]: gl.LINEAR,
-        [gl.TEXTURE_MIN_FILTER]: gl.LINEAR_MIPMAP_NEAREST
-      }]
+    return loadTextures(gl, {
+      urls: ['star.gif']
     })
     .then(textures => {
-      tStar = textures[0];
-
-      // Load all world objects
+      var stars = [];
       var numStars = 50;
       for (var i = 0; i < numStars; i++) {
-        scene.add(new Star(i / numStars * 5.0, i / numStars));
+        stars.push(new Star({
+          gl,
+          startingDistance: ((i / numStars) * 5.0),
+          rotationSpeed: (i / numStars),
+          texture: textures[0]
+        }));
       }
+      return {stars};
     });
   },
-  onRender() {
-    var camera = new PerspectiveCamera({
-      aspect: canvas.width / canvas.height
-    });
-
+  onRender: ({
+    gl, tick, aspect, stars
+  }) => {
     // Update Camera Position
-    var radTilt = tilt / 180 * Math.PI;
-    camera.position.set(0, Math.cos(radTilt) * zoom,
-                           Math.sin(radTilt) * zoom);
-    camera.update();
-    // Render all elements in the Scene
-    scene.render({camera});
+    const radTilt = tilt / 180 * Math.PI;
+    const cameraY = Math.cos(radTilt) * zoom;
+    const cameraZ = Math.sin(radTilt) * zoom;
 
-    scene.children.forEach(function(star) {
-      star.animate();
-    });
+    gl.clear(GL.COLOR_BUFFER_BIT | GL.DEPTH_BUFFER_BIT);
+
+    for (var i in stars) {
+      var uMVMatrix = Matrix4
+        .lookAt({eye: [0, cameraY, cameraZ]})
+        .multiplyRight(stars[i].matrix);
+
+      stars[i].render({
+        uMVMatrix,
+        uPMatrix: Matrix4.perspective({aspect})
+      });
+      stars[i].animate();
+    }
   }
 });
 
+function addControls({controlPanel} = {}) {
+  /* global document */
+  controlPanel = controlPanel || document.querySelector('.control-panel');
+  if (controlPanel) {
+    controlPanel.innerHTML = `
+  <p>
+    <a href="http://learningwebgl.com/blog/?p=1008" target="_blank">
+      Improving the code structure with lots of moving objects
+    </a>
+  <p>
+    The classic WebGL Lessons in luma.gl
+    `;
+  }
+}
+
+function addKeyboardHandler(canvas) {
+
+  addEvents(canvas, {
+    onKeyDown(e) {
+      switch (e.key) {
```

Comment:
I think it would be nice to briefly describe the key bindings in the text.This is probably overkill but could give a nice look:
<img width="66" alt="screen shot 2017-06-06 at 5 24 22 pm" src="https://user-images.githubusercontent.com/7025232/26857090-03dfae08-4add-11e7-85e7-bcd8296423bc.png">

```
        <table>
          <tbody>
            <tr>
              <td><div className="key">&#x2192;</div></td>
              <td>advance one frame (0.1s)</td>
            </tr>
            <tr>
              <td><div className="key">&#x2190;</div></td>
              <td>back up one frame (0.1s)</td>
            </tr>
            </tr>
          </tbody>
        </table>
```

With style as
```
 table {
    /* Give key table a little tighter row spacing */
    td {
      padding-top: 2px;
      padding-bottom: 2px;
      padding-left: 0px;
    }
    /* "keyboard key" style */
    .key {
        display: inline;
        display: inline-block;
        min-width: 1em;
        padding: .5em .6em .4em .6em;
        margin: 2px;
        font-weight: normal;
        font-size: 11px;
        text-align: center;
        text-decoration: none;
        line-height: 0.8rem;
        -moz-border-radius: .3em;
        -webkit-border-radius: .3em;
        border-radius: .3em;
        border: none;
        cursor: default;
        -moz-user-select: none;
        -webkit-user-select: none;
        user-select: none;
        background: #ffffff;
        border: 1px solid #ccc;
    }
  }
```